src/java/com/puppetlabs/puppetserver/pool/LockablePool.java

```diff
@@ -0,0 +1,25 @@
+package com.puppetlabs.puppetserver.pool;
+
+import java.util.concurrent.TimeUnit;
+
+public interface LockablePool<E> {
+    public void register(E e) throws InterruptedException;
+
+    public E borrowItem() throws InterruptedException;
+
+    public E borrowItemWithTimeout(long timout, TimeUnit unit) throws InterruptedException;
+
+    public void returnItem(E e) throws InterruptedException;
+
+    public void insertPill(E e) throws InterruptedException;
+
+    public void clear();
+
+    public int remainingCapacity();
+
+    public int size();
```

Comment:
Well, ideally, the only things that would be in the interface would be public methods that we really wanted to allow callers to use as part of the API, and if this were pure Java it's possible we'd leave those out of the interface and just make them `protected` methods that we called in the few spots where we need them for internal purposes, but clojure doesn't really play nicely with `protected` methods and it would be debatable whether that was a better way to do it even if clojure did play well.  So I think this is fine.